code/game/objects/structures/floodlight.dm

```diff
@@ -0,0 +1,322 @@
+/obj/machinery/power/floodlight
+	name = "floodlight"
+	icon = 'icons/obj/floodlight.dmi'
+	icon_state = "floodlight0_hatchopen0"
+	desc = "An industrial floodlight. It has units of power remaining."
+	density = 1
+	anchored = 0
+	var/status = 0
+	var/obj/item/weapon/stock_parts/cell/powerpack
+	var/datum/effect_system/spark_spread/sparks = new
+	var/health = 60
+	var/maxhealth = 60
+	var/broken = 0
```

Comment:
the vars broken and cover are useless, because /obj/machinery has a var called stat that covers both(and could cover more)